src/analyze/expr.rs

```diff
@@ -642,6 +666,8 @@ impl PureAnalyzer {
             (Type::Pointer(target, _), _) => ((**target).clone(), left, right),
             // i[p]
             (_, Type::Pointer(target, _)) => ((**target).clone(), right, left),
+            // already type error
+            (Type::Error, _) | (_, Type::Error) => return left,
```

Comment:
Should this split into `return left` and `return right`?

Reply:
Yes, that sounds like a good idea.

Reply:
6f2e58d93ef6856d887903c62b38d03e30c89e51